### DISCUSSION TITLE
Suggestion: Render Macro --> Render Partial

### DISCUSSION THREAD

**Discussion by hfloyd** [leekelleher/umbraco-contentment#299] [Idea]
With the conversation around deprecation of macros in Umbraco 13, I was thinking that it might make sense for there to be a "Render Partial" property editor which would operate similarly to the Render Macro editor. 

It could take a relative path to a PV file, and config data could be passed in via ViewData Key/Value pairs. Your standard values could automatically be included in that ViewData as well:

- "__propertyAlias" will be populated with the property's alias.
- "__propertyLabel" will be populated with the property's label (name).
- "__propertyCulture" will be populated with the currently selected culture.
- "__propertyDataTypeKey" will be populated with the Data Type's GUID value.
- "__propertyDescription" will be populated with the property's description.
- "__propertyHideLabel" will be the flag of whether the property's label is hidden or not.

![image](https://user-images.githubusercontent.com/1252593/213525736-fb7b2d13-f16c-441d-a7c5-bf9328f2f372.png)

**Comment by leekelleher** [comment]
Thanks @hfloyd, good suggestion! 👍 

I think I missed a discussion about Macros being deprecated in v13. 😬

**Reply by hfloyd** [reply]
Maybe it hasn't been officially announced yet... but as you know there have been conversations about "what to do about macros" for years now... and I think there is a general feeling that they aren't being used as much.

**Comment by hfloyd** [comment]
I was just thinking about this again today. I use "back-office only" macros on a few projects and thinking ahead to their next upgrades (v17 LTS), I'd hate to lose the functionality just because macros are gone. 

This morning I had a new idea - rather than using a Key/Value list for the parameters, perhaps it could be configured with a link to a Document Type, sort of like how Blocks in the RTE work. This would substitute a nice UI of a "property form" like the current "Render Macro" functionality uses the UI of macro parameters.

So, to configure, a Doctype would be specified (for the params) and a path to the partial view for rendering. I still think your set of standard "property" values could be available as ViewData automatically, so the partial view could access them without the developer having to configure them (like in the current Render Macro).

**Comment by hfloyd** [comment]
Hi @leekelleher! Looking at an upgrade project and this is back on my radar. If you think this would be helpful, I can take a crack at it and submit a PR. 😁 (If you don't think anyone else would want it, I can just make my own separate package. Just LMK what you think.)

**Reply by leekelleher** [reply]
Hi @hfloyd! 😀 So far, I haven't had any comments about Render Macro being defunct, but according to my [telemetry stats](https://leekelleher.com/umbraco/contentment/telemetry/), it wasn't that widely used. My gut feeling is that a Render Partial may be similar niche.

I'm open to a collaboration on this. Although, if you wanted to move faster on it, I'm also cool with you rolling your own package.

**Reply by hfloyd** [reply]
Ok, thanks! I have some other things to update to v17 compatibility first, but will come back to this. 😊

I appreciate your quick reply. ❤️ (See you soon at Codegarden!)
